### PR TITLE
feat(webserver): list config repositories from the repository_list endpoint

### PR DESCRIPTION
**Pull request by boxbeam**

### CODE REVIEW

**Comment by wsxiaoys** [comment]
cc @liangfung - is this sufficient for adapting the condig based repos?

**Comment by liangfung** [comment]
> cc @liangfung - is this sufficient for adapting the condig based repos?

Correspondingly, modifying the logic of `resolve repository entries` endpoint would be sufficient.

**Comment by boxbeam** [comment]
Will include screenshots when working

**Comment by boxbeam** [comment]
image
image

File: ee/tabby-schema/Cargo.toml [hunk 1]
```diff
@@ -27,6 +27,7 @@ tracing.workspace = true
 validator = { version = "0.16.1", features = ["derive"] }
 regex.workspace = true
 hash-ids.workspace = true
+tabby-git = { path = "../../crates/tabby-git" }
```
Comment (wsxiaoys) [inline ee/tabby-schema/Cargo.toml:30]: what's this for?

File: ee/tabby-schema/src/schema/mod.rs [hunk 2]
```diff
@@ -419,7 +423,28 @@ impl Query {
 
     async fn repository_list(ctx: &Context) -> Result<Vec<Repository>> {
         check_user(ctx).await?;
-        ctx.locator.repository().repository_list().await
+        let mut repositories = ctx.locator.repository().repository_list().await?;
+
+        repositories.extend(
```
Comment (wsxiaoys) [inline ee/tabby-schema/src/schema/mod.rs:428]: this should goes into service's `repository_list`

File: ee/tabby-schema/src/schema/mod.rs [hunk 3]
```diff
@@ -419,7 +423,28 @@ impl Query {
...
+            ctx.locator
+                .config()
+                .repositories
+                .iter()
+                .enumerate()
+                .map(|(index, repo)| {
+                    Ok(Repository {
+                        id: ID::new(format!("CONFIG_{}", index)),
```
Comment (wsxiaoys) [inline ee/tabby-schema/src/schema/mod.rs:436]: 1. the serialization / deserialization of config based id should be defined in `config.rs`
2. the format should be something like `config:${hashids encoded index}`

File: ee/tabby-webserver/src/service/mod.rs [hunk 4]
```diff
@@ -241,6 +246,10 @@ impl ServiceLocator for ArcServerContext {
     fn integration(&self) -> Arc<dyn IntegrationService> {
         self.0.integration.clone()
     }
+
+    fn config(&self) -> &Config {
+        &self.0.config
+    }
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/service/mod.rs:252]:
```suggestion
```

File: ee/tabby-schema/Cargo.toml (hunk 1 above)
Comment (boxbeam) [reply ee/tabby-schema/Cargo.toml:30]: Need to be able to read metadata (refs) from the repositories we're returning

**Pull request merged**